modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -283,6 +292,47 @@ public boolean checkStructEquivalency(BType actualType, BType expType) {
         return true;
     }
 
+    public boolean checkConnectorEquivalancy(BType actualType, BType expType) {
```

Comment:
Here we only need to check for action signatures? Please check how function types, struct types are matched.  I think we don't need to lookup all the action symbols for that. Just a thought

Reply:
Other than not expecting the same action order, I'm doing the same thing as struct equivalency right? I can't see a reason as to why we need to enforce same action order in both connectors, hence didn't expected them in same order

Reply:
When checking the connector type equivalency we need to consider the action signatures. We shouldn't use action symbol in this algorithm.

Reply:
yes got shall I do that change in separate PR?